tests/time_tests/c++/pipeline.h

```diff
@@ -0,0 +1,73 @@
+#include <chrono>
+#include <functional>
+
+#include <inference_engine.hpp>
+using namespace InferenceEngine;
+
+
+/**
+ * @brief Class response for encapsulating measure and measurements printing
+ *
+ * Current class measures only in scope of it's lifetime. In this case need
+ * to note that deletion of objects created before class creation may lead
+ * to negative values because of alignment on starting values.
+ * Also deletion of objects created in scope of class lifetime may decrease
+ * values computed on previous measure.
+ */
+class TimePipeline {
+private:
+    std::chrono::high_resolution_clock::time_point start;      // measure before run (will be used as baseline)
+
+    /**
+     * @brief Measures values at the current point of time
+     */
+    std::chrono::high_resolution_clock::time_point _measure();
+public:
+    /**
+     * @brief Constructs TimePipeline object and
+     *        measure values to use as baseline
+     */
+    // TODO: open file to dump statistics
+    TimePipeline(std::string statistics_path);
+
+    /**
+     * @brief Constructs TimePipeline object and
+     *        measure values to use as baseline
+     */
+    // TODO: close file using to dump statistics
+    ~TimePipeline();
+
+    /**
+     * @brief Measures values at the current point of time and
+     *        returns measurements aligned on a baseline
+     */
+    std::chrono::high_resolution_clock::time_point measure();
+
+    /**
+     * @brief Measures values and records aligned measurements using
+     *        provided identifier
+     */
+    // TODO: dump statistics to a file to emulate IPC while run IE pipeline several times
+    void record_measures(const std::string & id);
+
+    /**
+     * @brief Prepares string used for fast generation of file with references
+     */
+    // TODO: think about necessity
+    std::string get_reference_record_for_test(std::string test_name, std::string model_name,
+                                              std::string target_device);
+};
+
+
+int runPipeline(const std::string &statistics_path, const std::string &model, const std::string &device) {
+    // TODO: runPipeline may throw exceptions. In case of IPC need to handle exceptions via process status
+    TimePipeline(statistics_path);
+
+    Core ie;
+    CNNNetwork cnnNetwork = ie.ReadNetwork(model);
+    TimePipeline.record_measures("read_network");
+
+    ExecutableNetwork exeNetwork = ie.LoadNetwork(cnnNetwork, device);
+    TimePipeline.record_measures("first_time_inference");
+    return 0;
```

Comment:
The usage shoudl be

int runPipeline(const std::string &model, const std::string &device) {
    Core ie;
    auto start_time = start_measure()
    CNNNetwork cnnNetwork = ie.ReadNetwork(model);
    record_time("read_network", start_time);
...
}

Reply:
Implemented according to provided example